dotnet/src/SemanticKernel.Core/Planning/PlanExtensions.cs

```diff
@@ -68,13 +69,24 @@ public static string ToPlanString(this Plan plan, string indent = " ")
         return planString;
     }
 
+    /// <summary>
+    /// Returns decorated instance of <see cref="ISKFunction"/> with plan enabled instrumentation.
+    /// </summary>
+    /// <param name="plan">Instance of <see cref="Plan"/> to decorate.</param>
+    /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> to use for logging. If null, no logging will be performed.</param>
+    public static ISKFunction WithInstrumentation(this Plan plan, ILoggerFactory? loggerFactory = null)
+    {
+        return new InstrumentedPlan(plan, loggerFactory);
+    }
+
     /// <summary>
     /// Returns decorated instance of <see cref="IPlan"/> with enabled instrumentation.
     /// </summary>
     /// <param name="plan">Instance of <see cref="IPlan"/> to decorate.</param>
     /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> to use for logging. If null, no logging will be performed.</param>
+    [Obsolete("Use concrete class Plan WithInstrumentation instead")]
     public static IPlan WithInstrumentation(this IPlan plan, ILoggerFactory? loggerFactory = null)
     {
-        return new InstrumentedPlan(plan, loggerFactory);
+        throw new NotSupportedException("This method is obsolete, use concrete class Plan WithInstrumentation instead");
```

Comment:
If we will replace `Plan` type with `ISKFunction` interface in `InstrumentedPlan` class, it should be possible to keep `return new InstrumentedPlan(plan, loggerFactory);` line instead of throwing an exception. Otherwise, there are chances that user will receive runtime error, so in theory it will be breaking change.

Reply:
Can´t apply to this method, as InstrumentedPlan is not an `IPlan` anymore.

![image](https://github.com/microsoft/semantic-kernel/assets/19890735/d7f97b58-95c7-4ada-a881-21116bc7f218)

Reply:
Got it, makes sense!